adapters/handlers/rest/clusterapi/object_ttl.go

```diff
@@ -102,34 +111,58 @@ func (d *ObjectTTL) incomingDelete() http.Handler {
 			return
 		}
 
-		d.logger.WithFields(logrus.Fields{"action": "incoming_delete_expired_objects", "classes": len(body)}).Info("received request to delete expired objects")
+		d.logger.WithFields(logrus.Fields{
+			"action":  "object_ttl_deletions",
+			"classes": len(body),
+		}).Info("received request to delete expired objects")
 
 		// run the deletion in a separate goroutine to free up the HTTP handler immediately
 		enterrors.GoWrapper(func() {
 			// make sure to unlock the requestRunning flag when all deletions are done
 			defer d.requestRunning.Store(false)
+
+			start := time.Now()
+
+			ec := errorcompounder.NewSafe()
 			eg := enterrors.NewErrorGroupWrapper(d.logger)
+			eg.SetLimit(concurrency.TimesFloatGOMAXPROCS(d.config.ObjectsTTLConcurrencyFactor))
 
-			ec := errorcompounder.New()
-			for _, classPayload := range body {
-				className := classPayload.Class
+			objectsDeleted := atomic.Int32{}
+			onObjectsDeleted := func(count int32) { objectsDeleted.Add(count) }
 
-				idx, err := d.remoteIndex.IndexForIncomingWrite(context.Background(), className, 0)
-				if err != nil {
-					ec.Add(fmt.Errorf("get index for class %q: %w", className, err))
-					continue
+			for _, classPayload := range body {
+				collection := classPayload.Class
+				onCollectionError := func(err error) {
+					if err != nil {
+						ec.Add(fmt.Errorf("collection %q [%w]", collection, err))
```

Comment:
This error handling adds quite some complexity - should we just print any errors straight to the logger and not return much so we can avoid the error compounder? Any errors that currently appear are passed up the stack and then printed to the logger

I think errors would be the exception.

(Also fine to do an extra PR)